H2TauTau/python/proto/analyzers/H2TauTauTreeProducerBase.py

```diff
@@ -206,6 +207,62 @@ def fillExtraMetInfo(self, tree, event):
         self.fill(tree, 'pfmet_mt1', DiTau.calcMT(event.pfmet, event.leg1))
         self.fill(tree, 'pfmet_mt2', DiTau.calcMT(event.pfmet, event.leg2))
 
+    # TauSpinner information
+    def bookTauSpinner(self, tree):
+        self.var(tree, 'TauSpinnerWTisValid')
+        self.var(tree, 'TauSpinnerWT')
+        self.var(tree, 'TauSpinnerWThminus')
+        self.var(tree, 'TauSpinnerWThplus')
+        self.var(tree, 'TauSpinnerTauPolFromZ')
+        self.var(tree, 'TauSpinnerWRight')
+        self.var(tree, 'TauSpinnerWLeft')
+        self.var(tree, 'TauSpinnerIsRightLeft')
+
+    def fillTauSpinner(self, tree, event):
+        self.fill(tree, 'TauSpinnerWTisValid', event.TauSpinnerWTisValid )
+        self.fill(tree, 'TauSpinnerWT', float(event.TauSpinnerWT))
+        self.fill(tree, 'TauSpinnerWThminus', float(event.TauSpinnerWThminus))
+        self.fill(tree, 'TauSpinnerWThplus', float(event.TauSpinnerWThplus))
+        self.fill(tree, 'TauSpinnerTauPolFromZ', float(event.TauSpinnerTauPolFromZ))
+        self.fill(tree, 'TauSpinnerWRight', float(event.TauSpinnerWRight))
+        self.fill(tree, 'TauSpinnerWLeft', float(event.TauSpinnerWLeft  ))
+        self.fill(tree, 'TauSpinnerIsRightLeft', float(event.TauSpinnerIsRightLeft ))
+
+    def bookTopPtReweighting(self, tree):
+        self.var(tree, 'gen_top_1_pt')
+        self.var(tree, 'gen_top_2_pt')
+        self.var(tree, 'gen_top_weight')
+
+    def fillTopPtReweighting(self, tree, event):
+
+        ttbar = [p for p in event.genParticles if abs(p.pdgId())==6 and p.statusFlags().isLastCopy() and p.statusFlags().fromHardProcess()]
+
+        if self.cfg_comp.isMC and self.cfg_comp.name.find('TT')!=-1 and len(ttbar)==2:
```

Comment:
yes, it would be ...


2016-04-18 19:19 GMT+02:00 Jan Steggemann <notifications@github.com>:

> In H2TauTau/python/proto/analyzers/H2TauTauTreeProducerBase.py
> <https://github.com/steggema/cmgtools-lite/pull/10#discussion_r60097651>:
>
> > +        self.fill(tree, 'TauSpinnerWThplus', float(event.TauSpinnerWThplus))
> > +        self.fill(tree, 'TauSpinnerTauPolFromZ', float(event.TauSpinnerTauPolFromZ))
> > +        self.fill(tree, 'TauSpinnerWRight', float(event.TauSpinnerWRight))
> > +        self.fill(tree, 'TauSpinnerWLeft', float(event.TauSpinnerWLeft  ))
> > +        self.fill(tree, 'TauSpinnerIsRightLeft', float(event.TauSpinnerIsRightLeft ))
> > +
> > +    def bookTopPtReweighting(self, tree):
> > +        self.var(tree, 'gen_top_1_pt')
> > +        self.var(tree, 'gen_top_2_pt')
> > +        self.var(tree, 'gen_top_weight')
> > +
> > +    def fillTopPtReweighting(self, tree, event):
> > +
> > +        ttbar = [p for p in event.genParticles if abs(p.pdgId())==6 and p.statusFlags().isLastCopy() and p.statusFlags().fromHardProcess()]
> > +
> > +        if self.cfg_comp.isMC and self.cfg_comp.name.find('TT')!=-1 and len(ttbar)==2:
>
> Would this reweight TTH samples?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/steggema/cmgtools-lite/pull/10/files/0ae91bc91b8972190e5db65096f4e9afacd110ca#r60097651>
>



-- 
Yuta Takahashi
LHC - CMS experiment,
CERN (European Organization for Nuclear Research)